### DISCUSSION TITLE
Windows clients behind NAT on IKEv2 and EAP-TLS kick out each other

### DISCUSSION THREAD

**Discussion by joresp** [strongswan/strongswan#2031] [Q&A]
**Connecting two IKEv2 EAP Windows clients with certificates (behind NAT) to the VPN server causes them to kick out each other because they assign each other the same IP. This is of course, expected since the server cannot distinguish them.**

This problem never happened to me when using macOS or iOS as clients since these allow you to specify a "Local ID" and "Remote ID" for policy matching. I could just then match "leftid" and "rightid" in the configuration and would work great.

It's been days researching the documentation and the internet and **the only conclusion I've got is that using EAP-TLS you can never distinguish two Windows clients on the same IP**. Can someone please prove wrong? Is there any solution?

Here is my configuration:

```
# VPN for workstation
conn win-client-b8ab
    type=tunnel
    auto=start
    keyexchange=ikev2
    mobike=yes

    ike=aes128-sha1-modp1024,aes128-sha1-modp1536,aes128-sha1-modp2048,aes128-sha256-ecp256,aes128-sha256-modp1024,aes128-sha256-modp1536,aes128-sha256-modp2048,aes256-aes128-sha256-sha1-modp2048-modp4096-modp1024,aes256-sha1-modp1024,aes256-sha256-modp1024,aes256-sha256-modp1536,aes256-sha256-modp2048,aes256-sha256-modp4096,aes256-sha384-ecp384,aes256-sha384-modp1024,aes256-sha384-modp1536,aes256-sha384-modp2048,aes256-sha384-modp4096,aes256gcm16-aes256gcm12-aes128gcm16-aes128gcm12-sha256-sha1-modp2048-modp4096-modp1024,3des-sha1-modp1024!
    esp=aes128-aes256-sha1-sha256-modp2048-modp4096-modp1024,aes128-sha1,aes128-sha1-modp1024,aes128-sha1-modp1536,aes128-sha1-modp2048,aes128-sha256,aes128-sha256-ecp256,aes128-sha256-modp1024,aes128-sha256-modp1536,aes128-sha256-modp2048,aes128gcm12-aes128gcm16-aes256gcm12-aes256gcm16-modp2048-modp4096-modp1024,aes128gcm16,aes128gcm16-ecp256,aes256-sha1,aes256-sha256,aes256-sha256-modp1024,aes256-sha256-modp1536,aes256-sha256-modp2048,aes256-sha256-modp4096,aes256-sha384,aes256-sha384-ecp384,aes256-sha384-modp1024,aes256-sha384-modp1536,aes256-sha384-modp2048,aes256-sha384-modp4096,aes256gcm16,aes256gcm16-ecp384,3des-sha1!
    aggressive=no
    keyingtries=%forever

    ikelifetime=28800s
    lifetime=3600s

    dpddelay=60s
    dpdtimeout=120s
    dpdaction=clear

    eap_identity=%identity # For Windows to work

    # Server
    left=%any # The VPN gateway can have any IP
    leftsubnet=10.111.0.0/24 # Subnet of the local network
    leftcert=host-vpn.der # The certificate representing the server
    leftsendcert=always # For macOS and iOS clients to get the certificate without need to install it

    # Client
    right=%any # The client can have any IP
    rightsourceip=10.111.0.200/28 # Replace with reserved range for VPN clients.
    rightauth=eap-tls # For certificate authentication using EAP
```

**Comment by joresp** [comment]
Okay, so after a couple more hours of research I found the solution in the Github discussion #1817 by changing to EAP-MSCHAPv2 authentication, which uses the user name as one of the parameters to identity the connection:

So I changed these two lines in `ipsec.conf`:
```
eap_identity=%identity # For Windows to work
rightauth=eap-tls # For certificate authentication using EAP
```

...to this:
```
eap_identity=%any # For the identity to be asked before the MSCHAPv2 authentication
rightauth=eap-mschapv2 # For authentication using certificate, username and password
```

And I had to load the eap-identity plugin by adding it to `strongswan.conf`'s plugin object:

```
# The eap-identity module allows to request the identity before the EAP-MSCHAPv2 exchange is initiated.
eap-identity {
	load = yes
}
```

Also added individual users to `ipsec.secrets`

```
# Users for EAP-MSCHAPv2:
user1 : EAP 123456
user2 : EAP 12345678
```

**Good. From now on, as long as I authenticate using a different username for each connection, two connections will be maintained independently for each client.**

However, on my mind still I have the question if it was ever possible to maintain two different connections for two different clients on the same IP (NAT) using EAP-TLS as I wanted initially.

**Comment by tobiasbrunner** [comment]
Did your clients use the same certificate? Or at least the same subject DN in the certificates? Other than that, I can just say, read the log to see what's actually going on.
